### ISSUE TITLE
Misc rules from flake8-pie

### ISSUE THREAD

**Issue by sbdchd** [charliermarsh/ruff#1543]
There might be some overlap with these rules and pylint / flake8-simplify, but I think I eliminated most of the dupes / unnecessary rules:

- [ ] PIE810: [single-starts-ends-with](https://github.com/sbdchd/flake8-pie#pie810-single-starts-ends-with)
Instead of calling startswith or endswith on the same string for multiple prefixes, pass the prefixes as a tuple in a single startswith or endswith call.

```python
# error
foo.startswith("foo") or foo.startswith("bar")

# error
foo.endswith("foo") or foo.endswith("bar")

# error
foo.startswith("foo") or str.startswith(foo, "bar")

# ok
foo.startswith(("foo",  "bar"))

# ok
foo.endswith(("foo",  "bar"))

# ok
foo.startswith("foo") or foo.endswith("bar")
```

- [ ] PIE790: [no-unnecessary-pass](https://github.com/sbdchd/flake8-pie#pie790-no-unnecessary-pass)

`pass` is unnecessary when definining a class or function with an empty body.
```python
# error
class BadError(Exception):
    """
    some doc comment
    """
    pass

def foo() -> None:
    """
    some function
    """
    pass

# ok
class BadError(Exception):
    """
    some doc comment
    """

def foo() -> None:
    """
    some function
    """
```

- [ ] PIE794: [dupe-class-field-definitions](https://github.com/sbdchd/flake8-pie#pie794-dupe-class-field-definitions)

Finds duplicate definitions for the same field, which can occur in large ORM model definitions.

```python
# error
class User(BaseModel):
    email = fields.EmailField()
    # ...80 more properties...
    email = fields.EmailField()

# ok
class User(BaseModel):
    email = fields.EmailField()
    # ...80 more properties...
```

- [ ] [PIE796: prefer-unique-enums](https://github.com/sbdchd/flake8-pie#pie796-prefer-unique-enums)

By default the stdlib enum allows multiple field names to map to the same value, this lint requires each enum value be unique.

```python
# error
class Foo(enum.Enum):
    A = "A"
    B = "B"
    C = "C"
    D = "C"

# ok
class Foo(enum.Enum):
    A = "A"
    B = "B"
    C = "C"
    D = "D"
```

- [ ] [PIE800: no-unnecessary-spread](https://github.com/sbdchd/flake8-pie#pie800-no-unnecessary-spread)

Check for unnecessary dict unpacking.

```python
# error
{**foo, **{"bar": 10}}

# ok
{**foo, "bar": 10}
```

- [ ] [PIE804: no-unnecessary-dict-kwargs](https://github.com/sbdchd/flake8-pie#pie804-no-unnecessary-dict-kwargs)

As long as the keys of the dict are valid Python identifier names, we can safely remove the surrounding dict.

```python
# error
foo(**{"bar": True})

# ok
foo(bar=True)
foo(**buzz)
foo(**{"bar foo": True})
```

- [ ] [PIE807: prefer-list-builtin](https://github.com/sbdchd/flake8-pie#pie807-prefer-list-builtin)

`lambda: []` is equivalent to the builtin list

```python
# error
@dataclass
class Foo:
    foo: List[str] = field(default_factory=lambda: [])

# ok
@dataclass
class Foo:
    foo: List[str] = field(default_factory=list)
```

Do these all sound good / are there any I should leave out? I skipped a few different ones from the original package that were too specific / awkward.

**Comment by charliermarsh** [comment]
Yeah these are all quite nice. I can try to implement one of them tomorrow to lay the groundwork for the plug-in and we can run from there.

**Comment by charliermarsh** [comment]
(Thanks for this thoughtful collation.)

**Comment by charliermarsh** [comment]
I didn't realize you actually wrote that plugin, very cool.

**Comment by ljesparis** [comment]
I was passing by and was wondering if I can write some of these ? I'm actually learning rust, but, these plugins look easy enough.

I actually got around to developing the **prefer_unique_enums rule** last night 😅  and i made a mistake with a PR so i closed it 😓

**Comment by charliermarsh** [comment]
@ljesparis - Yes please, thanks!

**Comment by charliermarsh** [comment]
@sbdchd - Are you planning to do PIE810 to close this one out? Just trying to keep tabs on who's working on what. Thanks!

**Comment by sbdchd** [comment]
Yeah I can do that one as well
